s5_base/run.sh

```diff
@@ -39,250 +39,148 @@ for f in "$lm_train" "$prondict_orig" "$g2p_model"; do \
-        nb_oov=$(join -1 1 -2 1 data/$n/vocab_text_only.tmp <(awk '$2 ~ /[[:print:]]/ { print $2" "$1 }' data/$n/words.cnt | sort -k1,1) \
-        | sort | awk '{total = total + $2}END{print total}')
+          <(cut -f1 $prondict | sort -u) > data/$n/vocab_text_only.tmp
+        nb_oov=$(join -1 1 -2 1 data/$n/vocab_text_only.tmp <(awk '$2 ~ /[[:print:]]/ { print $2" "$1 }' \
+           data/$n/words.cnt | sort -k1,1) | sort | awk '{total = total + $2}END{print total}')
         oov=$(echo "scale=3;$nb_oov/$nb_tokens*100" | bc)
-        echo "The out of vocabulary rate for $n is: $oov" >> oov_rate || exit 1;
-    done
+        echo "The out of vocabulary rate for $n is: $oov" || exit 1;
+    done > oov_rate
 fi
 
 if [ $stage -le 4 ]; then
     if [ ! -d data/lang ]; then
         echo "Create the lexicon"
         [ -d $localdict ] && rm -r $localdict
         mkdir -p $localdict data/lang/log
-        $train_cmd --mem 4G data/lang/log/prep_lang.log \
-        local/prep_lang.sh \
-        $prondict        \
-        $localdict   \
-        data/lang
+        $train_cmd data/lang/log/prep_lang.log local/prep_lang.sh $prondict $localdict data/lang
     fi
 fi
 
 if [ $stage -le 5 ]; then
     echo "Preparing a pruned trigram language model"
     mkdir -p data/log
-    $train_cmd --mem 24G data/log/make_LM_3g.log \
-    local/make_LM.sh \
-    --order 3 --carpa false \
-    --min1cnt 20 --min2cnt 10 --min3cnt 2 \
-    $lm_train data/lang \
-    $localdict/lexicon.txt data \
-    || error 1 "Failed creating a pruned trigram language model"
-    
+    $train_cmd --mem 24G data/log/make_LM_3g.log local/make_LM.sh --order 3 --carpa false \
+      --min1cnt 20 --min2cnt 10 --min3cnt 2 $lm_train data/lang $localdict/lexicon.txt data \
+      || error 1 "Failed creating a pruned trigram language model";
+
     echo "Preparing an unpruned 4g LM"
-    $train_cmd --mem 32G data/log/make_LM_4g.log \
-    local/make_LM.sh \
-    --order 4 --carpa true \
-    --min1cnt 1 --min2cnt 0 --min3cnt 0 \
-    $lm_train data/lang \
-    $localdict/lexicon.txt data \
-    || error 1 "Failed creating an unpruned 4-gram language model"
-    
+    $train_cmd --mem 32G data/log/make_LM_4g.log local/make_LM.sh --order 4 --carpa true \
+      --min1cnt 1 --min2cnt 0 --min3cnt 0 $lm_train data/lang $localdict/lexicon.txt data \
+      || error 1 "Failed creating an unpruned 4-gram language model";
 fi
 
-# train a monophone system
+# Subsample train for faster start
 if [ $stage -le 6 ]; then
     echo "Make subsets of the training data to use for the first mono and triphone trainings"
-    
-    utils/subset_data_dir.sh data/train 40000 data/train_40k
-    utils/subset_data_dir.sh --shortest data/train_40k 5000 data/train_5kshort
-    utils/subset_data_dir.sh data/train 10000 data/train_20k
+    utils/subset_data_dir.sh data/train 5000 data/train_5k
+    utils/subset_data_dir.sh data/train 10000 data/train_10k
```

Comment:
Why did you reduce the subsets so much? Will it even learn anything with such small subsets?

Reply:
I left the 5k and 10k as they were. I removed 40k subset because it will make the script fail on teen data as there is < 40k files originally. This way the script is fully operational on both general and teen.